backend/auth/internal/server/server.go

```diff
@@ -94,7 +94,14 @@ func (s *APIServer) SetupRoutes() {
 // Run starts the API server and listens for incoming requests.
 func (s *APIServer) Run() {
 	s.SetupRoutes()
-	s.Router.Run(":" + os.Getenv(config.AuthServerPort))
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = os.Getenv(config.AuthServerPort)
+	}
+	if port == "" {
+		port = "3001"
+	}
+	s.Router.Run(":" + port)
```

Comment:
Could you give us a brief explanation of how this works? Is the "PORT" variable something you're setting in a .env file in GCP?

The value of `config.AuthServerPort` is "AUTH_SERVER_PORT" which is being set by the docker-compose file

https://github.com/cs467-spring2025-discussion-app/capstone-discussion-app/blob/6a1f480f292943b0861ebf7815a5bf9cdae82c5b/docker-compose.yml#L28

Could you use "AUTH_SERVER_PORT" instead of "PORT" wherever you're setting the values for these variables, or is this a special case? I'm not attached to "AUTH_SERVER_PORT" itself, it's just a more a specific variable name than "PORT", whose value we might confuse across the different service.

Reply:
GCP/Cloud Run specifically sets the PORT variable...if that's not present (because we aren't running on GCP and running locally), it will use the docker-compose setup and set AUTH_SERVER_PORT, and then the fallback is port 3001 (which is what I think you had before). Per your question, it's a special case in the sense that this is how GCP/Cloud Run works to my understanding.

Reply:
> GCP/Cloud Run specifically sets the PORT variable

If Cloud Run is forcing our hand, (seems like it according to [this](https://cloud.google.com/run/docs/container-contract#port)) I'd like to handle the case where our services might want to share an environment (quick local testing where I don't want to spin up the docker container for an isolated environment)  and we have clashing "PORT" values.

Can we alter the priority order so that if "AUTH_SERVER_PORT" is set (which we just won't set in GCP), it will run the server on that value rather than any "PORT" value? (and also enforce explicitly setting a port value rather than falling back to a default). So:

```suggestion
	var port string
	if val, isSet := os.LookupEnv(config.AuthServerPort); isSet {
		port = val
	} else if val, isSet := os.LookupEnv("PORT"); isSet {
		port = val
	}
	if port == "" {
		log.Fatal().Err(fmt.Errorf("Auth server port not set")).Msg("Exiting due to empty port")
	}
	log.Info().Str("port", port).Msg("Starting auth server")
	s.Router.Run(":" + port)
}
```

We'd have to add `"github.com/rs/zerolog/log"` (EDIT: and `"fmt"`) to the list of imports for this to compile.

Reply:
Yes - I think that should work!